server/routes.go

```diff
@@ -1108,10 +1110,16 @@ func (s *Server) GenerateRoutes() http.Handler {
 	}
 	config.AllowOrigins = envconfig.Origins()
 
+	m, err := telemetry.InitMetrics()
+	if err != nil {
+		slog.Warn(fmt.Sprintf("Metrics initialization failed with %s", err))
```

Comment:
slog is a structured logger, I'd suggest leaning on that (removing the need to format should be more performant, too) 
```suggestion
		slog.Warn("Metrics initialization failed", "err", err)
```